src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnAutoInsertHandler.cs

```diff
@@ -70,7 +74,17 @@ public async Task<DocumentOnAutoInsertResponseItem> HandleRequestAsync(DocumentO
             }
 
             var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
-            if (projectionResult == null || projectionResult.LanguageKind != RazorLanguageKind.Html)
+            if (projectionResult == null || projectionResult.LanguageKind == RazorLanguageKind.Razor)
```

Comment:
I can definitely get a little "helper-happy" but when I see three if's in a row that all return null I feel like this might be a spot for `IsApplicableProjectionResult(projectionResult);` or something similar.